testutil/fstest/compare.go

```diff
@@ -35,3 +38,17 @@ func CheckDirectoryEqual(d1, d2 string) error {
 
 	return nil
 }
+
+// CheckDirectoryEqualWithApplier compares directory against applier
+func CheckDirectoryEqualWithApplier(a Applier, d string) error {
+	// TODO: cache applied
```

Comment:
If this function is called multiple times against identical `Applier`, caching the content of `applied` would be better for optimal performance.

Of course, ideally we should not even create `applied` directory. I'll try to redesign this.